### PR TITLE
Update kotlintest to version 3.3.1

### LINKED ISSUE #1330: Update KotlinTest to 3.3.X
The new IDE plugin only works on 3.3.X!

https://proandroiddev.com/introducing-the-kotlintest-intellij-plugin-cf8005e9177a

### PR DESCRIPTION
**Pull request by LordRaydenMK**
Law updated because the function has to be suspended.

NumberMonoid updated because nesting is not supported and I had to create some generators.

Moore updated because the default `toString` implementation breaks kotlin-reflect.

Fixes #1330

thanks @pakoito for the help :smile:

### CODE REVIEW

**Comment by pakoito** [comment]
The build pauses after

```
arrow.extensions.NumberMonoidTest > should semigroup with the instance passed - int PASSED
arrow.extensions.NumberMonoidTest > should semigroup with the instance passed - float PASSED
arrow.extensions.NumberMonoidTest > should semigroup with the instance passed - double PASSED
arrow.extensions.NumberMonoidTest > should semigroup with the instance passed - long PASSED
arrow.extensions.NumberMonoidTest > should semigroup with the instance passed - short PASSED
```

File: modules/core/arrow-extras-data/src/main/kotlin/arrow/data/Moore.kt [hunk 1]
```diff
@@ -13,5 +13,7 @@ data class Moore<E, V>(val view: V, val handle: (E) -> Moore<E, V>) : MooreOf<E,
 
   fun extract(): V = view
 
+  override fun toString() = "Moore $view"
```
Comment (pakoito) [inline modules/core/arrow-extras-data/src/main/kotlin/arrow/data/Moore.kt]: `Moore(view=$view, handle=(E) -> Moore<E, V>)`

Comment (LordRaydenMK) [reply modules/core/arrow-extras-data/src/main/kotlin/arrow/data/Moore.kt]: done

File: modules/core/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt [hunk 2]
```diff
@@ -15,10 +15,10 @@ import io.kotlintest.properties.Gen
 import java.util.concurrent.TimeUnit
 
 fun Gen.Companion.short(): Gen<Short> =
-        Gen.int().filter { it >= Short.MIN_VALUE && it <= Short.MAX_VALUE }.map { it.toShort() }
+        Gen.choose(Short.MIN_VALUE.toInt(), Short.MAX_VALUE.toInt()).map { it.toShort() }
 
 fun Gen.Companion.byte(): Gen<Byte> =
-        Gen.int().filter { it >= Byte.MIN_VALUE && it <= Byte.MAX_VALUE }.map { it.toByte() }
```
Comment (pakoito) [inline modules/core/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt]: 👍

**Pull request merged**